_posts/2023-02-01-dask-on-flyte.md

```diff
@@ -0,0 +1,151 @@
+
+After registration with Flyte, the workflow can be started from the UI:
+
+<img src="/images/dask-flyte-workflow.png" alt="Dask workflow in the Flyte UI" style="max-width: 100%;" width="100%" />
+
+## Why use the `dask` plugin for Flyte?
+
+At a first glance, both Flyte and `dask` look similar in what they are trying to achieve. Both tools can create a DAG out of user functions, take care of connecting in- and outputs, etc. The major conceptual difference is in how both achieve these. While `dask` has long-lived workers to run tasks, a Flyte task is a designated Kubernetes Pod. This creates a significant overhead in task-runtime.
+
+While `dask` tasks have an overhead of around one millisecond ([docs](https://distributed.dask.org/en/stable/efficiency.html#use-larger-tasks)), spinning up a new Kubernetes pod takes on the order of seconds. In addition, due to the long-lived nature of the `dask` workers, `dask` can optimize the DAG to run tasks operating on the same piece of data to run on the same node to avoid the need to serialize the outputs between workers ("shuffling"). Due to the ephemeral nature of Flyte `task`s, this is not possible and task outputs will be serialized to a blob storage instead.
+
+So given all oft the downsides above, why use Flyte? Flyte does not aim at replacing tools such as `dask` or Apache Spark, but rather adds an orchestration layer on top. Sure, while you can run workloads directly in Flyte - think training a model on a single GPU for instance - Flyte offers [a lot of integrations](https://flyte.org/integrations) for other common data processing tools.
+
+In this particular case, a `dask` Flyte task will manage the `dask` cluster lifecycle. When the task is started from the UI, Flyte will first create a dedicated `dask` cluster (made up of Kubernetes pods) which will then be used to execute the user code. This has the advantage of different tasks being able to use different docker images with different dependencies, whilst always ensuring that the dependencies of the client, scheduler and workers are the same.
+
+## What prerequisites are required to run `dask` tasks in Flyte?
+
+- The Kubernets cluster needs to have the [dask operator](https://kubernetes.dask.org/en/latest/operator.html) installed.
+- Flyte needs to be `>=1.3.0`
+- The `dask` plugin needs to be enabled in the Flyte (propeller) config ([docs](https://docs.flyte.org/en/latest/deployment/plugins/k8s/index.html#specify-plugin-configuration))
+- The Python environment within the docker image associated task needs to have `flytekitplugins-dask` installed
+
+## How do things work under the hood?
+
+_Note: The following should only be informative and should not be required for pure users of the plugin, it might be helpful for easier debugging though_
```

Comment:
```suggestion
_Note: The following is for reference only and is not necessary for users who only use the plugin. However, it could be useful for easier debugging._
```